usm_rest_api/v1/views/tasks.py

```diff
@@ -0,0 +1,224 @@
+        status, failed = usm_rest_utils.create_gluster_cluster(probeList)
+        log.debug("Failed Nodes from create_ceph_cluster %s" % failed)
+    elif cluster_data['cluster_type'] == usm_rest_utils.CLUSTER_TYPE_CEPH:
+        current_task.update_state(state='SETUP_CEPH_CLUSTER')
+        status, failed = usm_rest_utils.create_ceph_cluster(
+            probeList, cluster_data, minionIds)
+        log.debug("Failed Nodes from create_ceph_cluster %s" % failed)
+
+    # if status is False then cluster creation Failed, no need to
+    # proceed further. Otherwise check failed list and merge to the \
+    # failed nodes list
+    if not status:
+        log.critical("Cluster creation Failed %s and Nodelist %s" %
+                     (cluster_data, nodelist))
+        return {'status': 'ALL_NODES_FAILURE', 'failednodes': str(nodelist)}
+    else:
+        # merge the failed list
+        failedNodes += failed
+    log.debug("Done. Executed the commands on nodes")
+    successNodes = [item for item in probeList if item not in failedNodes]
+    log.debug("Successful Nodes %s" % successNodes)
+    log.debug("Failed Nodes %s" % failedNodes)
+    current_task.update_state(state='UPDATE_CLUSTER_STATUS')
+    try:
+        #
+        # Update the status of the successful Nodes in DB
+        # HOST_STATUS_INACTIVE = 1
+        # HOST_STATUS_ACTIVE = 2
+        #
+        usm_rest_utils.update_host_status(
+            successNodes, usm_rest_utils.HOST_STATUS_ACTIVE)
+        # Update the status of Cluster to Active if atlest two nodes
+        # are added successfully for gluster cluster
+        # Update the status of Cluster to Active if atlest one nodes
+        # are added successfully for ceph cluster
+        # STATUS_INACTIVE = 1
+        # STATUS_ACTIVE_NOT_AVAILABLE = 2
+        # STATUS_ACTIVE_AND_AVAILABLE = 3
+        #
+        if (len(successNodes) >= 1 and cluster_data['cluster_type']
+                == usm_rest_utils.CLUSTER_TYPE_CEPH) or \
+                (len(successNodes) >= 2 and cluster_data['cluster_type'] ==
+                 usm_rest_utils.CLUSTER_TYPE_GLUSTER):
+            usm_rest_utils.update_cluster_status(
+                str(cluster_data['cluster_id']),
+                usm_rest_utils.STATUS_ACTIVE_AND_AVAILABLE)
+
+    except Exception, e:
+        log.exception(e)
+
+    if noOfNodes == len(failedNodes):
+        log.critical("Cluster creation Failed %s and Nodelist %s" %
+                     (cluster_data, nodelist))
+        return {'status': 'ALL_NODES_FAILURE', 'failednodes': str(failedNodes)}
+    elif failedNodes:
+        log.critical("Cluster creation partially Failed %s and Nodelist %s" %
+                     (cluster_data, failedNodes))
+        return {'status': 'PARTIAL_FAILURE', 'failednodes': str(failedNodes)}
+
+    return states.SUCCESS
+
+
+@shared_task
+def createHost(data):
+    log.debug("Inside createGlusterHost Async Task %s" % data)
```

Comment:
debug message doesn't agree with the function. Is it createHost or createGlusterHost